File: patterns-java/src/test/java/me/freedom4live/patterns/java/singleton/SingletonTest.java
package me.freedom4live.patterns.java.singleton;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

class SingletonTest {

    @Test
    void getInstances_hasInstances_theSame() {
        //arrange
        Singleton previous = Singleton.getInstance();

        for (int i = 0; i < 1000; i++) {
            //act
            Singleton current = Singleton.getInstance();

            //assert
            assertNotNull(current);
            assertEquals(previous, current);
            assertEquals(previous.getName(), current.getName());

            previous = current;
        }
    }

}

File: patterns-kotlin/src/test/kotlin/me/freedom4live/patterns/kotlin/singleton/SingletonTest.kt
package me.freedom4live.patterns.kotlin.singleton

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.Test

internal class SingletonTest {

    @Test
    fun getInstance_noCreation_theSame() {
        //arrange
        var previous = Singleton

        repeat(1000) {
            //act
            val current = Singleton


            //assert
            assertEquals(previous, current)
            assertEquals(previous.name, current.name)
            assertSame(previous, current)

            previous = current
        }
    }

}
